types/connect/index.d.ts

```diff
@@ -1,94 +1,163 @@
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /**
+         * Register an "error-handling" middleware. These are middleware where the
+         * function takes exactly 4 arguments. When a middleware passes an error to `next`, the app will
+         * proceed to look for the error middleware that was declared after that middleware and invoke it,
+         * skipping any error middleware above that middleware and any non-error middleware below.
+         * @param fn A plain function, a Node.js HTTP server instance or another Connect app instance.
+         */
+        use(fn: (
+            /** The error that occurred. */ err: any,
+            /** The Node.js http request object. */ req: Request,
+            /** The Node.js http response object. */ res: http.ServerResponse,
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /**
+         * Mount an "error-handling" middleware on the specified route.
+         * @param route An optional path string that is matched against the beginning of the incoming request URL.
+         * This allows for basic routing.
+         *
+         * The route is always terminated at a path separator (`/`) or a dot (`.`) character. This means the
+         * given routes `/foo/` and `/foo` are the same and both will match requests with the URLs
+         * `/foo`, `/foo/`, `/foo/bar`, and `/foo.bar`, but not match a request with the URL `/foobar`.
+         *
+         * The route is matched in a case-insensitive manner.
+         * @param fn A plain function, a Node.js HTTP server instance or another Connect app instance.
+         */
+        use(route: string, fn: (
+            /** The error that occurred. */ err: any,
+            /** The Node.js http request object. */ req: Request,
+            /** The Node.js http response object. */ res: http.ServerResponse,
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /** Register a middleware. */
         use(fn: HandleFunction): Server;
+        /** Mount a middleware on the specified route. */
         use(route: string, fn: HandleFunction): Server;
 
         /**
-        * Handle server requests, punting them down
-        * the middleware stack.
-        *
-        * @private
-        */
-        handle(req: http.IncomingMessage, res: http.ServerResponse, next: Function): void;
+         * Calling the function will run the middleware stack against the given Node.js http request (`req`)
+         * and response (`res`) objects. An optional function `out` can be provided that will be called if the
+         * request (or error) was not handled by the middleware stack.
+         */
+        handle(req: http.IncomingMessage, res: http.ServerResponse, out?: NextFunction): void;
```

Comment:
> An optional function out can be provided that will be called if the request (or error) was not handled by the middleware stack.
[Code](https://github.com/senchalabs/connect/blob/master/index.js#L127) (lines 127, 151) confirms that the out param is optional and called with err as parameter.